### PR TITLE
Adding support for URLs

### PR DESCRIPTION
**Pull request by lpetre**
- [ ] @danbrakeley 
- [ ] @lpetre

### CODE REVIEW

File: params.go [hunk 1]
```diff
@@ -45,6 +46,18 @@ type StringSliceParam struct {
 	value []string
 }
 
+type URLParam struct {
+	baseParam
+	def   string
```
Comment (lpetre) [inline params.go:51]: This is different than the rest of the types. I can't decide if I want:

``` golang
    RedisURL      = params.URL("redis.url", &url.URL{Host: "localhost:6379", Scheme: "redis"})
```

or 

``` golang
    RedisURL      = params.URL("redis.url", "redis://localhost:6379")
```

The issue with the first one is verbosity. The issue with the second one is this second `def` member, and the parsing of `def` into a url.URL, which can fail!

Comment (danbrakeley) [inline params.go:51]: How does time.Duration processing work?

Comment (danbrakeley) [inline params.go:51]: Right, it parses it only in the config file, otherwise it wants to be passed something of type time.Duration.

Comment (lpetre) [inline params.go:51]: This is how onion does Durations. Maybe the better move here is to add support for URLs to onion?
https://github.com/fzerorubigd/onion/blob/v2/onion.go#L351

**Pull request closed without merging**